services/pull/conflicts.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2025 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package pull
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	issues_model "code.gitea.io/gitea/models/issues"
+	"code.gitea.io/gitea/modules/git"
+	"code.gitea.io/gitea/modules/git/gitcmd"
+	"code.gitea.io/gitea/modules/gitrepo"
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/util"
+)
+
+// checkPullRequestMergeableAndUpdateStatus checks whether a pull request is mergeable and updates its status accordingly.
+// It uses 'git merge-tree' if supported by the Git version, otherwise it falls back to using a temporary repository.
+// This function updates the pr.Status, pr.MergeBase and pr.ConflictedFiles fields as necessary.
+// The pull request parameter may not be created yet in the database, so do not assume it has an ID.
+func checkPullRequestMergeableAndUpdateStatus(ctx context.Context, pr *issues_model.PullRequest) error {
```

Comment:
I have told you, you need to clearly test this function's behavior, but not keep copy-pasting or polluting other unrelated tests